### PR TITLE
[SIG-3984] Incident overview TypeScript

### PR DESCRIPTION
**Pull request by MauritsR**
## Changes

- Refactors incident overview list to TypeScript

### CODE REVIEW

File: src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.tsx [hunk 1]
```diff
@@ -30,86 +41,41 @@ const getDaysOpen = (incident) => {
   const hasDaysOpen =
     incident.status && !statusesWithoutDaysOpen.includes(incident.status.state)
   const start = hasDaysOpen && parseISO(incident.created_at)
-  return hasDaysOpen ? -differenceInCalendarDays(start, new Date()) : '-'
+  return start ? -differenceInCalendarDays(start, new Date()) : '-'
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.tsx:44]: If this change doesn't break the corresponding test, is the test coverage sufficient?

File: src/utils/__tests__/fixtures/incidents.json [hunk 2]
```diff
@@ -58,23 +60,43 @@
...
-    "assigned_user_email": "user1@example.com"
+    "assigned_user_email": "user1@example.com",
+    "has_attachments": false,
+    "has_children": false,
+    "has_parent": false,
+    "notes": [
+      {
+        "created_by": null,
+        "text": "Automatische e-mail bij registratie van de melding is verzonden aan de melder."
+      }
```
Comment (janjaap) [inline src/utils/__tests__/fixtures/incidents.json:80]: I could be mistaken, but I'm not seeing the `notes` prop in the endpoint's output: https://api.data.amsterdam.nl/api/swagger/?url=/signals/swagger/openapi.yaml#/default/get_signals_v1_private_signals_.

**Review by janjaap — COMMENTED** [review]
A couple of comments. Also, might be good to tackle the TS warning in `useBuildGetter.ts`.

File: src/utils/__tests__/fixtures/incidents.json (hunk 2 above)
Comment (MauritsR) [reply src/utils/__tests__/fixtures/incidents.json:80]: Good point - I'll ask to have this property added to the docs.

Comment (MauritsR) [reply src/utils/__tests__/fixtures/incidents.json:80]: This property will be added to the Swagger docs

File: src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.test.tsx [hunk 3]
```diff
@@ -14,85 +15,67 @@ import districts from 'utils/__tests__/fixtures/districts.json'
...
-      container.querySelector('tr:nth-child(1) td:nth-child(6)')
-    ).toHaveTextContent(/^Centrum$/)
-    expect(
-      container.querySelector('tr:nth-child(1) td:nth-child(7)')
-    ).toHaveTextContent(incidents[0].category.sub)
-    expect(
-      container.querySelector('tr:nth-child(1) td:nth-child(8)')
-    ).toHaveTextContent(incidents[0].status.state_display)
+      screen.getByRole('row', {
+        name: '1668 - 03-12-2018 10:41 Centrum Wegsleep Afgehandeld Staalstraat 3B 1011JJ Amsterdam',
```
[the remark this answers is not in the document]
Comment (MauritsR) [reply src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.test.tsx]: Changed it to a template string, where values displayed directly from the data are interpolated. 

Other values I've left the same, to make it clear how the formatted strings should be displayed (e.g. time)

File: src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.tsx (hunk 1 above)
Comment (MauritsR) [reply src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.tsx]: We're not verifying that `created_at` is indeed a date. In my opinion we shouldn't need to validate the format of backend data - what do you think?

Comment (janjaap) [reply src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.tsx]: We shouldn't, no. But, if we're relying on a valid date format, we should have a test for it, I guess.

File: src/hooks/api/useBuildGetter.ts [hunk 4]
```diff
@@ -12,7 +12,8 @@ export const useBuildGetter = <T, U extends Array<unknown> = Array<any>>(
         const [url, params] = paramBuilder(...args)
         return genericGet(url, params)
       },
-    []
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [genericGet]
```
Comment (janjaap) [inline src/hooks/api/useBuildGetter.ts:16]: Why omit `paramBuilder` from the list of deps?

**Review by janjaap — APPROVED** [review]

File: src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.tsx (hunk 1 above)
Comment (MauritsR) [reply src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.tsx]: Added the test

**Pull request merged**